decentraland/unity-explorer
# fix: Multiple duplicated requests for remote profiles

## What does this PR change?
Fix #2142 

We have detected a huge amount of duplicated requests to the Profile each time any of the other avatars around us changes something in his profile: Backpack, Passport, etc.

![image](https://github.com/user-attachments/assets/7a3267a3-705a-431e-8926-953d793f2ac8)

![image](https://github.com/user-attachments/assets/346729fb-eb03-4427-af33-1df6a0cad57d)

In order to avoid this, this PR introduces the detection of all the duplicated `RemoteAnnouncement` downloadings (`RemoteProfiles` class) in order to send only one per bunch.

## How to test the changes?
1. Launch the explorer.
2. Stay in some place with several users around.
3. Make the other users change anything related to their profiles: Change wearables and/or modify the profile information from the passport.
4. Check that each change on the other users is being applied correctly: See the wearables change visually, open their passports and see that their infos are correctly updated, etc.
5. Make other users go to another scene and come back to the current one, and check that the avatar loads correctly: correct wearables and correct passport info.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
